RecoHGCal/TICL/interface/commons.h

```diff
@@ -40,6 +56,44 @@ namespace ticl {
   // verbosity levels for ticl algorithms
   enum VerbosityLevel { None = 0, Basic, Advanced, Expert, Guru };
 
+  inline int returnClusterType(DetId& lc_seed, const hgcal::RecHitTools& rhtools_) {
+    auto layer_number = rhtools_.getLayerWithOffset(lc_seed);
+    auto thickness = rhtools_.getSiThickIndex(lc_seed);
+    auto isEELayer = (layer_number <= rhtools_.lastLayerEE(false));
+    auto isScintillator = rhtools_.isScintillator(lc_seed);
+    auto isFine = (layer_number <= rhtools_.lastLayerEE(false) + 7);
```

Comment:
`7`?
No hard coded numbers: these will cause troubles in the long term, especially if related to the geometry.
On more general grounds, maybe all this should be moved to `RecHittools`?
In the end, the type of cluster is derived from its seed which is a `RecHit`.

Reply:
Also, why returning `-1`? If, for whatever reason, we reach that statement, all downstream code has to check if the type returned is valid.
A nice `assert` is safer, IMHO. At least you die where you were supposed to.

Reply:
@waredjeb Do you want to comment here?

Reply:
@srimanob Yes, we discussed it offline, I moved this piece to the RecHitTools and added an assert to avoid returning -1

Reply:
Sorry, it's still confusing why we have a 7 there? Can this be clarified?